### PR TITLE
fix(mechanics): Don't pass through wormholes if fleet jumping until escorts are ready

### PR DESCRIPTION
**Pull request by Koranir**
**Bug fix**

This PR addresses the bug/feature described in issue #10065

## Summary
This adds the same kind of mechanism as existing fleet jumps were for jumping on going through wormholes.

### CODE REVIEW

**Review by TomGoodIdea — APPROVED** [review]
Code is ok, but I can't test it, because the bug doesn't trigger for me.

File: source/AI.cpp [hunk 1]
```diff
@@ -116,6 +116,24 @@ namespace {
 		}
 		return true;
 	}
+	bool EscortsReadyToLand(const Ship &ship)
```
Comment (RisingLeaf) [inline source/AI.cpp:119]:
```suggestion

	bool EscortsReadyToLand(const Ship &ship)
```

**Review by RisingLeaf — COMMENTED** [review]
Looks good

**Review by RisingLeaf — APPROVED** [review]

File: source/AI.cpp [hunk 2]
```diff
@@ -117,6 +117,25 @@ namespace {
...
+		bool shipIsYours = ship.IsYours();
+		const Government *gov = ship.GetGovernment();
+		for(const weak_ptr<Ship> &ptr : ship.GetEscorts())
+		{
+			shared_ptr<const Ship> escort = ptr.lock();
+			// Skip escorts which are not player-owned and not escort mission NPCs.
+			if(!escort || (shipIsYours && !escort->IsYours() && (!escort->GetPersonality().IsEscort()
+				|| gov->IsEnemy(escort->GetGovernment()))))
+				continue;
+			if(escort->IsDisabled() || escort->CanBeCarried())
```
Comment (tibetiroka) [inline source/AI.cpp:131]: Since fighters can land on wormholes, I don't think they should be excluded.
```suggestion
			if(escort->IsDisabled())
```

**Review by tibetiroka — CHANGES_REQUESTED** [review]

**Review by tibetiroka — APPROVED** [review]

**Pull request merged**